### PR TITLE
Add image autobumper documentation

### PR DESCRIPTION
**Pull request by KacperMalachowski**
**Description**

For now, the image autobumper is a tool used internally in test-infra repository. 
The documentation will be used internally for now as well, but we might offer the image-autobumper as a service in the future.

Changes proposed in this pull request:

- Add the documentation for the image-autobumper

**Related issue(s)**

Fixes: #12056

### CODE REVIEW

**Review by Sawthis — DISMISSED** [review]

File: cmd/image-autobumper/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+Image Autobumper is a tool for automatically updating the version of a Docker image in a GitHub repository. It is designed to automatically detected the latest version of the image and update the version in the repository based on the provided configuration.
+
+Key features:
+* Automatically detects the latest version of the image
+* Automatically detects places in the file where image urls are defined
+* Automatically updates the version of the image in the files
+
+## Quickstart Guide
+
+You can use Image Autobumper in your GitHub workflow to update the version of a Docker image in a GitHub repository. It autoamtically detects yaml files (both `.yaml` and `.yml`) in the repository. You can configure the tool to update the version of the image in other files with `extraFiles` option in the configuration file.
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:12]: This is mainly repetition of what you've already said.

File: cmd/image-autobumper/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+[Here](https://github.com/kyma-project/test-infra/blob/main/.github/workflows/autobump-images.yml) is an example of a GitHub workflow updating the version of a Docker image using Image Autobumper.
+[Here](https://github.com/kyma-project/test-infra/blob/main/configs/autobump-config/test-infra-autobump-config.yaml) is an example of a configuration file for Image Autobumper.
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:14]:
```suggestion
Use Image Autobumper in your GitHub workflow to detect YAML files (both `.yaml` and `.yml`) in the repository. With the `extraFiles' option in the configuration file, you can also configure the tool to update the version of the image in other files.
See the following examples:
* [GitHub workflow updating the Docker image version using Image Autobumper](https://github.com/kyma-project/test-infra/blob/main/.github/workflows/autobump-images.yml)
* [Configuration file for Image Autobumper](https://github.com/kyma-project/test-infra/blob/main/configs/autobump-config/test-infra-autobump-config.yaml)
```

File: cmd/image-autobumper/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## Supported events
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:16]:
```suggestion
## Supported Events
```

File: cmd/image-autobumper/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Image Autobumper supports all events that trigger a GitHub Workflow, except the `pull_request` event. That limitation is caused by requirement to fetch the `kyma-bot` token, which is used to access the forked repository. The `pull_request` event does not provide the OIDC token required to authenticate against Google Cloud Secret Manager.
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:18]:
```suggestion
Image Autobumper supports all events that trigger a GitHub workflow except the **pull_request** event. That limitation is caused by the requirement to fetch the `kyma-bot` token, which is used to access the forked repository. The **pull_request** event does not provide the OIDC token required to authenticate against Google Cloud Secret Manager.
```

File: cmd/image-autobumper/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+## Reusable workflow reference
+
+The workflow that uses Image Autobumper reusable workflow must use the exact reference to the reusable workflow. The value of the `uses` key must be `kyma-project/test-infra/.github/workflows/image-autobumper.yml@main`.
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:22]:
```suggestion
The workflow that uses Image Autobumper reusable workflow must use the exact reference to the reusable workflow. The value of the **uses** key must be `kyma-project/test-infra/.github/workflows/image-autobumper.yml@main`.
```

File: cmd/image-autobumper/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+```yaml
+uses: kyma-project/test-infra/.github/workflows/image-builder.yml@main
+```
+
+> [!IMPORTANT]
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:28]:
```suggestion
> [!WARNING]
```

File: cmd/image-autobumper/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+> Using different references to the reusable workflow will result in an error during the workflow execution.
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:29]:
```suggestion
> Using different references to the reusable workflow results in an error during the workflow execution.
```

File: cmd/image-autobumper/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+## Reusable workflow inputs
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:31]:
```suggestion
## Reusable Workflow Inputs
```

File: cmd/image-autobumper/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+The Image Autobumper reusable workflow accepts inputs to parametrize the build process. See the accepted inputs description in the [image-autobumper reusable workflow](/.github/workflows/image-autobumper.yml) file.
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:33]:
```suggestion
The Image Autobumper reusable workflow accepts inputs to parametrize the build process. See the accepted inputs description in the [`image-autobumper` reusable workflow](/.github/workflows/image-autobumper.yml) file.
```

File: cmd/image-autobumper/README.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+You can use Image Autobumper in your GitHub workflow to update the version of a Docker image in a GitHub repository. It autoamtically detects yaml files (both `.yaml` and `.yml`) in the repository. You can configure the tool to update the version of the image in other files with `extraFiles` option in the configuration file.
+[Here](https://github.com/kyma-project/test-infra/blob/main/.github/workflows/autobump-images.yml) is an example of a GitHub workflow updating the version of a Docker image using Image Autobumper.
+[Here](https://github.com/kyma-project/test-infra/blob/main/configs/autobump-config/test-infra-autobump-config.yaml) is an example of a configuration file for Image Autobumper.
+
+## Supported events
+
+Image Autobumper supports all events that trigger a GitHub Workflow, except the `pull_request` event. That limitation is caused by requirement to fetch the `kyma-bot` token, which is used to access the forked repository. The `pull_request` event does not provide the OIDC token required to authenticate against Google Cloud Secret Manager.
+
+## Reusable workflow reference
```
Comment (IwonaLanger) [inline cmd/image-autobumper/README.md:20]:
```suggestion
## Reusable Workflow Reference
```

File: cmd/image-autobumper/README.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
+# Image Autobumper
+
+Image Autobumper is a tool for automatically updating the version of a Docker image in a GitHub repository. It is designed to automatically detected the latest version of the image and update the version in the repository based on the provided configuration.
```
[the remark this answers is not in the document]
Comment (KacperMalachowski) [reply cmd/image-autobumper/README.md]: Moved information into Key Features section

File: cmd/image-autobumper/README.md (hunk 1 above)
Comment (KacperMalachowski) [reply cmd/image-autobumper/README.md]: Removed repetetition, at least tried to remove all of them.

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**